### DISCUSSION TITLE
Share your TorchSharp Checkpoints at HuggingFace!

### DISCUSSION THREAD

**Discussion by GeorgeS2019** [xamarin/TorchSharp#1277] [General]
# Update 20th April 2024

Reference: https://github.com/dotnet/TorchSharp/issues/713#issuecomment-2059584658
https://huggingface.co/yueyinqiu/vision-TorchSharp/tree/main

![Screenshot_2024-04-20-11-19-01-605-edit_com microsoft emmx](https://github.com/dotnet/TorchSharp/assets/49812372/2a9e01fc-7949-4441-b4bd-290880548e39)

# Update 2nd April 2024
<details>
<summary>2nd April 2024: <strong>stable-diffusion-1-5-torchsharp</summary>
https://huggingface.co/williamlzw/stable-diffusion-1-5-torchsharp

![image](https://github.com/dotnet/TorchSharp/assets/49812372/eec55d55-2358-4e6e-9527-9975d6f6618a)
</details>

**Comment by GeorgeS2019** [comment]
@yueyinqiu 
One strong motivation for TorchSharp is that the .Net community is deprived of basic core framework for deep AI

Deep AI for vision is more biased support

We need equally support into Deep AI for Audio and more importantly Deep NLP

It would be great if you could do the same for TorchVision at huggingface to TorchAudio and eventually Torchtext

**Reply by yueyinqiu** [reply]
There seems to be no pre-trained models provided by torchaudio/torchtext. Have I missed something?

**Reply by GeorgeS2019** [reply]
There are only limited ones in TorchSharp: @kaiidams 

Most Deep AI developers are more familiar with vision than audio. We have a bias.
https://pytorch.org/audio/stable/models.html
https://github.com/pytorch/audio

**Reply by GeorgeS2019** [reply]
Not all models are consistent with the latest trends in Deep Audio. 

I would not consider those (FIRST) which are earlier than 2020

This (2020) remains still very relevant.
https://pytorch.org/audio/stable/generated/torchaudio.models.Wav2Vec2Model.html#torchaudio.models.Wav2Vec2Model

**Reply by GeorgeS2019** [reply]
Most of the TorchText models are **NOW** far more up-to-date in HuggingFace.
https://pytorch.org/text/main/models.html#

The utilities needed to work on them are scattered in different places. 

Even Microsoft dotnet team has challenges locating them and put them in one place
